src/skibidi_orm/migration_engine/db_config/base_config.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Callable
+
+
+class ConfigSingleton(type):
+    _instances: dict[Any, Any] = {}
+
+    def __call__(cls, *args: Any, **kwargs: Any) -> Any:
+        if cls not in cls._instances:
+            instance = super().__call__(*args, **kwargs)
+            cls._instances[cls] = instance
+        return cls._instances[cls]
+
+    @staticmethod
+    def instance_exists(class_name: Any) -> bool:
+        return class_name in ConfigSingleton._instances
+
```

Comment:
I don't know if custom metaclass is the way to go. Seems unnecessary, maybe we should go with normal inheritance.

Reply:
idk how to do it without two classes where one makes each config singleton, and 2nd one (parent) forbids instantiating Config of two different DB engines.